packages/block-editor/src/components/global-styles/shadow-panel-components.js

```diff
@@ -123,3 +124,47 @@ function renderShadowToggle() {
 		);
 	};
 }
+
+function ClearShadowButton( { onClear } ) {
+	return (
+		<Button
+			className="block-editor-global-styles__clear-shadow"
+			variant="tertiary"
+			onClick={ onClear }
+		>
+			{ __( 'Clear' ) }
+		</Button>
+	);
+}
+
+export function getShadowPresets( settings ) {
```

Comment:
If we didn't care about global styles, we could write this function much more simply.

I haven't tested it, but we should be able to write it like this: Additionally, memorize the return value and use an `EMPTY_ARRAY` variable.

```javascript
export function getShadowPresets( settings ) {
	if ( ! settings?.shadow ) {
		return EMPTY_ARRAY;
	}

	const defaultPresetsEnabled = settings?.shadow?.defaultPresets;
	const { default: defaultShadows, theme: themeShadows } = settings?.shadow;
	const unsetShadow = {
		name: __( 'Unset' ),
		slug: 'unset',
		shadow: 'none',
	};

	return useMemo(
		() => [
			unsetShadow,
			...( themeShadows || EMPTY_ARRAY ),
			...( ( defaultPresetsEnabled && defaultShadows ) || EMPTY_ARRAY ),
		],
		[ defaultPresetsEnabled, defaultShadows, themeShadows ]
	);
}
```

Reply:
updated the function with similar code structure and wrapped entire function call inside `useMemo`